src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,738 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+
+                    // Many-Many relationships are marked as Referencing relationships because the linking table insertion can happen only when records have been successfully created in both the
+                    // entities involved in the relationship.
+                    // In M:N relationship, both the entities are referenced entities and the linking table is the referencing table. 
+                    // So, populating referencing fields is performed only for 1:N relationships.    
+                    if (currentEntityRelationshipMetadata is not null
+                        && currentEntityRelationshipMetadata.TargetEntityToFkDefinitionMap.TryGetValue(relatedEntityName, out List<ForeignKeyDefinition>? referencingEntityFKDefinitions)
+                        && !referencingEntityFKDefinitions.IsNullOrEmpty())
+                    {
+                        ForeignKeyDefinition referencingEntityFKDefinition = referencingEntityFKDefinitions[0];
+
+                        if (referencingRelationshipMultipleCreateStructure.IsLinkingTableInsertionRequired)
+                        {
+                            PopulateReferencingFields(sqlMetadataProvider, referencingRelationshipMultipleCreateStructure, referencingEntityFKDefinition, multipleCreateStructure.CurrentEntityCreatedValues, isLinkingTable: true, entityName);
+                        }
+                        else
+                        {
+                            PopulateReferencingFields(sqlMetadataProvider, referencingRelationshipMultipleCreateStructure, referencingEntityFKDefinition, multipleCreateStructure.CurrentEntityCreatedValues, isLinkingTable: false, entityName);
+                        }
+                    }
+
+                    PerformDbInsertOperation(context, node.Value, sqlMetadataProvider, referencingRelationshipMultipleCreateStructure);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Helper method to extract the primary key fields from all the fields of the entity.
+        /// </summary>
+        /// <param name="sqlMetadataProvider">SqlMetadaProvider object for the given database</param>
+        /// <param name="entityName">Name of the entity</param>
+        /// <param name="entityFields">All the fields belonging to the entity</param>
+        /// <returns>Primary Key fields</returns>
+        private static Dictionary<string, object?> FetchPrimaryKeyFields(ISqlMetadataProvider sqlMetadataProvider, string entityName, Dictionary<string, object?> entityFields)
+        {
+            Dictionary<string, object?> pkFields = new();
+
+            SourceDefinition sourceDefinition = sqlMetadataProvider.GetSourceDefinition(entityName);
+            foreach (string primaryKey in sourceDefinition.PrimaryKey)
+            {
+                if (sqlMetadataProvider.TryGetExposedColumnName(entityName, primaryKey, out string? name) && !string.IsNullOrWhiteSpace(name))
```

Comment:
Is defining an empty/whitespace exposed column name even allowed?

Reply:
Good catch, removed